app/views/NewServerView/index.tsx

```diff
@@ -1,167 +1,79 @@
-	connecting: boolean;
-	previousServer: string | null;
-}
-
-interface INewServerViewState {
-	text: string;
-	certificate: string | null;
-	serversHistory: TServerHistoryModel[];
-	showBottomInfo: boolean;
-}
-
-interface ISubmitParams {
-	fromServerHistory?: boolean;
-	username?: string;
-}
-
-class NewServerView extends React.Component<INewServerViewProps, INewServerViewState> {
-	constructor(props: INewServerViewProps) {
-		super(props);
-		this.setHeader();
-
-		this.state = {
-			text: '',
-			certificate: null,
-			serversHistory: [],
-			showBottomInfo: true
-		};
-		EventEmitter.addEventListener('NewServer', this.handleNewServerEvent);
-		BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
-		if (isAndroid) {
-			Keyboard.addListener('keyboardDidShow', () => this.handleShowKeyboard());
-			Keyboard.addListener('keyboardDidHide', () => this.handleHideKeyboard());
-		}
-	}
-
-	componentDidMount() {
-		this.queryServerHistory();
-	}
-
-	componentWillUnmount() {
-		EventEmitter.removeListener('NewServer', this.handleNewServerEvent);
-		BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
-		if (isAndroid) {
-			Keyboard.removeAllListeners('keyboardDidShow');
-			Keyboard.removeAllListeners('keyboardDidHide');
-		}
-
-		const { previousServer, dispatch } = this.props;
-		if (previousServer) {
-			dispatch(serverFinishAdd());
-		}
-	}
-
-	componentDidUpdate(prevProps: Readonly<INewServerViewProps>) {
-		if (prevProps.connecting !== this.props.connecting) {
-			this.setHeader();
-		}
-	}
-
-	setHeader = () => {
-		const { previousServer, navigation, connecting } = this.props;
-		if (previousServer) {
-			return navigation.setOptions({
-				headerTitle: I18n.t('Add_server'),
-				headerLeft: () =>
-					!connecting ? (
-						<HeaderButton.CloseModal navigation={navigation} onPress={this.close} testID='new-server-view-close' />
-					) : null
-			});
-		}
-
-		return navigation.setOptions({
-			headerShown: false
-		});
-	};
-
-	handleBackPress = () => {
-		const { navigation, previousServer } = this.props;
+import { useAppSelector } from '../../lib/hooks';
+import useServersHistory from './hooks/useServersHistory';
+import useCertificate from './hooks/useCertificate';
+import useConnectServer from './hooks/useConnectServer';
+import { INewServerViewProps } from './definitions';
+import completeUrl from './utils/completeUrl';
+import CertificatePicker from './components/CertificatePicker';
+import styles from './styles';
+
+const NewServerView = ({ navigation }: INewServerViewProps) => {
+	const dispatch = useDispatch();
+	const { colors } = useTheme();
+	const { previousServer, connecting, failure } = useAppSelector(state => ({
+		previousServer: state.server.previousServer,
+		connecting: state.server.connecting,
+		failure: state.server.failure
+	}));
+
+	const { control } = useForm({ mode: 'onChange', defaultValues: { workspaceUrl: '' } });
+
+	const [text, setText] = useState<string>('');
```

Comment:
Going to mention this on the migration to hooks as well.
`useForm` should be in charge of managing the form state.